### PR TITLE
Setting Administrator Permissions via OIDC

### PR DESCRIPTION
**Pull request by clintonb**
Users updated via OIDC can now be marked as administrators. These administrators will have access to the admin portal as well as permission to view all courses.

@rocha @dsjen

### CODE REVIEW

File: analytics_dashboard/analytics_dashboard/backends.py [hunk 1]
```diff
@@ -81,6 +81,9 @@ def get_user_details(self, response):
         if locale:
             details[u'language'] = _to_language(response['locale'])
 
+        # Set superuser bit if the provider determines the user is an administrator
+        details[u'is_superuser'] = response.get('administrator', False)
```
Comment (rocha) [inline analytics_dashboard/analytics_dashboard/backends.py]: This propagates to the `django.contrib.auth` `User` model, correct?

**Comment by rocha** [comment]
:rocket:

One little question, other than that, it looks good to me.

File: analytics_dashboard/analytics_dashboard/backends.py (hunk 1 above)
Comment (clintonb) [inline analytics_dashboard/analytics_dashboard/backends.py]: Correct, but to our custom user model, `analytics_dashboard.models.User` which is a subclass of the default Django user model.

Comment (rocha) [inline analytics_dashboard/analytics_dashboard/backends.py]: Oh yeah, Django 1.6. Nice. Maybe we should add a comment that `get_user_details` is used by `social_auth` to automagically populate the user model.

**Comment by clintonb** [comment]
@rocha Please see the last commit. The `is_staff` bit is needed to enable access to the admin portal. `is_superuser` simply grants permissions (but not access to the admin portal).

**Comment by rocha** [comment]
Ah, good catch. :rocket:

**Comment by rocha** [comment]
Those django auth names are a bit confusing, even more given that we also have our own notions of "staff".

**Pull request merged**